app/src/main/java/org/mozilla/fenix/home/collections/CollectionViewHolder.kt

```diff
@@ -0,0 +1,194 @@
+
+@Composable
+private fun Collection(
+    collection: TabCollection,
+    expanded: Boolean,
+    menuItems: List<CollectionMenuItemData>,
+    interactor: CollectionInteractor,
+) {
+    var isMenuShown by remember { mutableStateOf(false) }
+    val isExpanded by remember { mutableStateOf(expanded) }
+
+    Card(
+        modifier = Modifier
+            .semantics(mergeDescendants = true) {}
+            .clickable { interactor.onToggleCollectionExpanded(collection, !isExpanded) }
+            .height(48.dp),
+        shape = if (isExpanded) expandedCollectionShape else collapsedCollectionShape,
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 5.dp, // This needs to match the elevation of the TabInCollectionViewHolder for the shadow.
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxHeight(),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Icon(
+                painter = painterResource(R.drawable.ic_tab_collection),
+                contentDescription = null,
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 8.dp // (24.dp - 16.dp) hardcoded in ExpandableListHeader
+                ),
+                tint = Paint().apply {
+                    color = Color(collection.getIconColor(LocalContext.current))
+                    blendMode = BlendMode.SrcIn
+                }.color
+            )
+
+            ExpandableListHeader(
+                headerText = collection.title,
+                expanded = isExpanded,
+            ) {
+                if (isExpanded) {
+                    Row {
+                        IconButton(
+                            onClick = {
+                                interactor.onCollectionShareTabsClicked(
+                                    collection
+                                )
+                            }
+                        ) {
+                            Icon(
+                                painter = painterResource(R.drawable.ic_share),
+                                contentDescription = stringResource(R.string.share_button_content_description),
+                                tint = FirefoxTheme.colors.textPrimary
+                            )
+                        }
+
+                        IconButton(
+                            onClick = {
+                                isMenuShown = !isMenuShown
+                                interactor.onCollectionMenuOpened()
+                            }
+                        ) {
+                            Icon(
+                                painter = painterResource(R.drawable.ic_menu),
+                                contentDescription = stringResource(
+                                    R.string.collection_menu_button_content_description
+                                ),
+                                tint = FirefoxTheme.colors.textPrimary
```

Comment:
I believe this should be `FirefoxTheme.colors.iconPrimary`

Reply:
Thank you, will change to that.